Support newer OpenSSL

Building with OpenSSL 1.1.1c fails due to undefined variables:
```
/usr/bin/c++  -DGUID_LIBUUID -DXEUS_EXPORTS -Dxeus_EXPORTS -I/builddir/build/BUILD/xeus-0.23.1/include -isystem /usr/include/pgm-5.2  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC   -Wunused-parameter -Wextra -Wreorder -march=native -std=gnu++14 -o CMakeFiles/xeus.dir/src/xauthentication.cpp.o -c /builddir/build/BUILD/xeus-0.23.1/src/xauthentication.cpp
/builddir/build/BUILD/xeus-0.23.1/src/xauthentication.cpp: In function 'const EVP_MD* xeus::asevp(const string&)':
/builddir/build/BUILD/xeus-0.23.1/src/xauthentication.cpp:111:27: error: 'EVP_mdc2' was not declared in this scope; did you mean 'EVP_md2'?
  111 |             {"hmac-mdc2", EVP_mdc2},
      |                           ^~~~~~~~
      |                           EVP_md2
/builddir/build/BUILD/xeus-0.23.1/src/xauthentication.cpp:121:9: error: could not convert '{{"hmac-md5", EVP_md5}, {"hmac-sha1", EVP_sha1}, {"hmac-mdc2", <expression error>}, {"hmac-ripemd160", EVP_ripemd160}, {"hmac-blake2b512", EVP_blake2b512}, {"hmac-blake2s256", EVP_blake2s256}, {"hmac-sha224", EVP_sha224}, {"hmac-sha256", EVP_sha256}, {"hmac-sha384", EVP_sha384}, {"hmac-sha512", EVP_sha512}}' from '<brace-enclosed initializer list>' to 'const std::map<std::__cxx11::basic_string<char>, const evp_md_st* (*)()>'
  121 |         };
      |         ^
      |         |
      |         <brace-enclosed initializer list>
```
Fedora has some compat package for 1.0.x, but it's orphaned and soon to be deleted.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,7 +49,7 @@ message(STATUS "xeus binary version: v${XEUS_BINARY_VERSION}")
 set(nlohmann_json_REQUIRED_VERSION 3.2.0)
 set(xtl_REQUIRED_VERSION 0.5)
 set(cppzmq_REQUIRED_VERSION 4.3.0)
-set(zeromq_REQUIRED_VERSION 4.2.3)
+set(zeromq_REQUIRED_VERSION 4.2.5)
 
 find_package(nlohmann_json ${nlohmann_json_REQUIRED_VERSION} REQUIRED)
 find_package(xtl ${xtl_REQUIRED_VERSION} REQUIRED)

--- a/src/xauthentication.cpp
+++ b/src/xauthentication.cpp
@@ -108,7 +108,8 @@ namespace xeus
         static const std::map<std::string, const EVP_MD*(*)()> schemes = {
             {"hmac-md5", EVP_md5},
             {"hmac-sha1", EVP_sha1},
-            {"hmac-mdc2", EVP_mdc2},
+            // MDC2 is disabled by default unless enable-mdc2 is specified
+            // {"hmac-mdc2", EVP_mdc2},
             {"hmac-ripemd160", EVP_ripemd160},
 #if OPENSSL_VERSION_NUMBER >= 0x10100000L
             {"hmac-blake2b512", EVP_blake2b512},

